### PR TITLE
Speed up GKS annotations and add interval based annotation

### PR DESCRIPTION
**Pull request by theferrit32**
- Builds on #556 
- Adds functions that annotate a variant table with most of the VA/VRS structure, within hail.
- gnomad_gks_batch fills in the missing objects in vanilla python using `ga4gh.vrs` and `ga4gh.core` modules after collecting the VRS and VA frequency structs from the hail table into Python dicts

### CODE REVIEW

**Comment by theferrit32** [comment]
Commit 6d1f4ad was to make this mergeable on top of #556

File: gnomad/resources/grch38/gnomad.py
Comment (mattsolo1) [inline gnomad/resources/grch38/gnomad.py:297]: @theferrit32 thank you

**Review by matren395 — COMMENTED** [review]
Kyle F walked through all the changes with me Friday the 7th, everything looks humane and I definitely understand the need for it.

**Comment by mattsolo1** [comment]
Thank you Daniel! Who customarily presses the merge button in this repo? @theferrit32?

**Comment by ch-kr** [comment]
@klaricch is still planning to review this pull request but likely won't get to the review until next week due to other responsibilities / some upcoming OOO time. @mattsolo1 do you and Phil need this PR merged to continue working on the API (can you use this branch until Kristen has had time to review/approve)?

**Comment by mattsolo1** [comment]
Sounds good. 

It's not blocking us, no. Just nice to have some closure for @theferrit32 😃 

Note this PR isn't even one going into main... it's a PR into an existing PR.

File: gnomad/resources/grch38/gnomad.py
Comment (klaricch) [inline gnomad/resources/grch38/gnomad.py:473]:
```suggestion
    :param coverage_ht: Table consisting of coverage data or string 'auto' to automatically lookup the publicly available gnomAD coverage ht. If None, coverage will not be annotated. Default is 'auto'.
```

File: gnomad/utils/annotations.py
Comment (klaricch) [inline gnomad/utils/annotations.py:1729]: why use hl.format here rather than fstrings, such as "f"{variant}.{group_id.upper()}"?

File: gnomad/utils/annotations.py
Comment (klaricch) [inline gnomad/utils/annotations.py:1794]: does something still need to be filled in here?

File: gnomad/utils/annotations.py
Comment (klaricch) [inline gnomad/utils/annotations.py:1620]:
```suggestion
    Add GKS VRS variant annotation to a Table.
```

File: gnomad/utils/annotations.py
Comment (klaricch) [inline gnomad/utils/annotations.py:1654]: i thought the desired output was a dictionary, not an annotated ht?

File: gnomad/utils/annotations.py
Comment (klaricch) [inline gnomad/utils/annotations.py:1622]: why can't location id be added with what was used previously:

File: gnomad/utils/annotations.py
Comment (klaricch) [inline gnomad/utils/annotations.py:1541]: could provide more detail here on how the dictionary is expected to be structured

File: gnomad/utils/annotations.py
Comment (klaricch) [inline gnomad/utils/annotations.py:1539]:
```suggestion
    have the digest-based id computed. Compute it and assign it to .location._id.
```

File: gnomad/utils/annotations.py
Comment (klaricch) [inline gnomad/utils/annotations.py:1565]:
```suggestion
    :param ht: Table containing VRS annotation under 'vrs'.
```

File: gnomad/utils/annotations.py
Comment (klaricch) [inline gnomad/utils/annotations.py:1566]:
```suggestion
    :param export_tmpfile: File path to export the output Table to.
```

File: gnomad/utils/annotations.py
Comment (klaricch) [inline gnomad/utils/annotations.py:1567]:
```suggestion
    :param computed_tmpfile: File path to write the updated rows to, which is then imported as a Table.
```

File: gnomad/utils/annotations.py
Comment (klaricch) [inline gnomad/utils/annotations.py:1592]:
```suggestion
                    # Serialize outputs to JSON and write to TSV
```

File: gnomad/utils/annotations.py
Comment (klaricch) [inline gnomad/utils/annotations.py:1610]: infer build rather than hardcoding

File: gnomad/utils/annotations.py
Comment (klaricch) [inline gnomad/utils/annotations.py:1559]:
```suggestion
    Compute sequence location digest-based id for a hail variant Table.
```

**Review by klaricch — CHANGES_REQUESTED** [review]
A lot of the code within add_gks_va and add_gks_vrs is very similar to what's already in other functions such as get_gks, so if anything changes about VRS it would have to be edited in multiple locations. Can we create constants/functions with the repetitive code?

File: gnomad/resources/grch38/gnomad.py
Comment (ch-kr) [reply gnomad/resources/grch38/gnomad.py:461]: I like this idea, will add

File: gnomad/utils/annotations.py
Comment (ch-kr) [reply gnomad/utils/annotations.py:1541]: added based on https://github.com/theferrit32/gnomad_methods/blob/3523684d048d4a64fd089f48e8232f3129d73651/gnomad/utils/annotations.py#L1364; let me know how this looks and if the types I added are right!

File: gnomad/utils/annotations.py
Comment (ch-kr) [reply gnomad/utils/annotations.py:1622]: @theferrit32

File: gnomad/utils/annotations.py
Comment (ch-kr) [reply gnomad/utils/annotations.py:1626]: @theferrit32

File: gnomad/utils/annotations.py
Comment (ch-kr) [reply gnomad/utils/annotations.py:1729]: not sure, but I vote for fstrings, so will add

File: gnomad/utils/annotations.py
Comment (theferrit32) [reply gnomad/utils/annotations.py:1610]: Does this still need to be done?

File: gnomad/utils/annotations.py
Comment (theferrit32) [reply gnomad/utils/annotations.py:1622]: The location id can't be added here because these are added has hail expressions and they can't call into normal python functions like the SequenceLocation init and `ga4gh_identify`. That step is deferred to after the rows are `collect()`ed. The code flow is a little confusing, and because of this complexity around not being able to compute the location id in a hail expression, which I was not anticipating when I first started. Because of that, it might be better to do all the vrs/va stuff after the `collect()` call. The filtering done in `gnomad_gks_batch` is only based on the locus, so it doesn't need the vrs/va stuff to be in the table itself. 

@klaricch @ch-kr Maybe we can discuss this next week.

File: gnomad/utils/annotations.py
Comment (theferrit32) [reply gnomad/utils/annotations.py:1626]: I was trying to see if the conversion to JSON at different steps affected the overall performance, but I didn't notice a difference. I think the `.vrs` struct isn't actually used anywhere later, only `.vrs_json`. Maybe this behavior should be made consistent between `add_gks_vrs` and `add_gks_va`. The `gks_va_freq_dict` field added by `add_gks_va` is added by a struct but is latler converted to JSON in `gnomad_gks_batch`

File: gnomad/utils/annotations.py
Comment (theferrit32) [reply gnomad/utils/annotations.py:1686]: Since `get_gks` function was split into a `add_gks_va` and `add_gks_vrs`, I think this is not needed. The decision about what whether only vrs should be created is made in `gnomad_gks_batch`.

File: gnomad/utils/annotations.py
Comment (theferrit32) [reply gnomad/utils/annotations.py:1729]: It is using `hl.format` because the `ht.gnomad_id` is a hail `StringExpression`, not a `str`

**Pull request merged**
